src/hooks/useProjects.ts

```diff
@@ -16,6 +16,8 @@ export const useProjects = (record?: { projectId?: number }) => {
   useEffect(() => {
     if (availableProjects && record?.projectId) {
       setSelectedProject(availableProjects.find((project) => project.id === record.projectId));
+    } else {
```

Comment:
@karthikbtf I was wrong about the race condition, this is why it wasn't refreshing correctly in the previous PR